webapp/pages/profile/_id/_slug.vue

```diff
@@ -403,6 +406,12 @@ export default {
       if (type === 'following') this.followingCount = count
       if (type === 'followedBy') this.followedByCount = count
     },
+    async showOrChangeChat(roomID) {
```

Comment:
In my opinion i don't like method names like doAOrDoB(), 
here I would prefer the name openChat or openSmallChat if it is specific to the SmallChat.
```suggestion
    async openChat(roomID) {
```
```suggestion
    async openSmallChat(roomID) {
```